subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ValidationActions.java

```diff
@@ -110,9 +113,81 @@ public void doValidate(String propertyName, Object values, TaskValidationContext
         }
     };
 
-    private static void validateNotInReservedFileSystemLocation(TaskValidationContext context, File location) {
+    private static void reportMissingInput(TaskValidationContext context, String kind, String propertyName, File input) {
+        context.visitPropertyProblem(problem -> {
+            String lowerKind = kind.toLowerCase();
+            problem.withId(ValidationProblemId.INPUT_DOES_NOT_EXIST)
+                .forProperty(propertyName)
+                .reportAs(ERROR)
+                .withDescription(() -> lowerKind + " '" + input + "' doesn't exist")
```

Comment:
The message currently is something like this:
```
Type 'DefaultTask': property 'myProperty' specifies directory '<path>' which does not exist. Input directories must exist...
```

That sounds a little bit strange to me.

Maybe something like this would be better:
```
Type 'DefaultTask': property 'myProperty' directory '<path>' doesn't exist. An input is missing...
```

Reply:
I don't understand your comment: in the message I changed, it doesn't say "specifies" anymore

Reply:
Sorry, I switched the current message an my suggestion in the comment. I fixed that now.